src/python/bot/tasks/progression_task.py

```diff
@@ -210,6 +213,32 @@ def _save_fixed_range(testcase_id, min_revision, max_revision):
   task_creation.request_bisection(testcase, 'fixed')
 
 
+def _store_testcase_for_regression_testing(testcase, testcase_file_path):
+  """Stores reproduction testcase for future regression testing in corpus
+  pruning task."""
+  if testcase.open:
+    # Store testcase only after the crash is fixed.
+    return
+
+  if not testcase.bug_information:
+    # Only store crashes with bugs associated with them.
+    return
+
+  fuzz_target = data_handler.get_fuzz_target(testcase.overridden_fuzzer_name)
+  if not fuzz_target:
+    # No work to do, only applicable for engine fuzzers.
+    return
+
+  corpus = corpus_manager.FuzzTargetCorpus(fuzz_target.engine,
+                                           fuzz_target.project_qualified_name())
+  regression_testcase_url = os.path.join(
+      corpus.get_regressions_corpus_gcs_url(),
+      os.path.basename(testcase_file_path))
+  storage.copy_file_to(testcase_file_path, regression_testcase_url)
+  logs.log('Successfully stored testcase for regression testing: ' +
```

Comment:
nit: move this under `if storage.copy_file_to(...):`

Reply:
can also log a warning / error in case of a failure

Reply:
Done!